### PR TITLE
Fix clippy warnings

### PR DESCRIPTION
**Pull request by rethab**
some spring cleaning :)

### CODE REVIEW

File: src/uu/cut/src/cut.rs [hunk 1]
```diff
@@ -535,40 +535,36 @@ pub fn uumain(args: impl uucore::Args) -> i32 {
         matches.value_of(options::CHARACTERS),
         matches.value_of(options::FIELDS),
     ) {
-        (Some(byte_ranges), None, None) => {
```
Comment (rethab) [inline src/uu/cut/src/cut.rs:538]: no changes here, only line wrapping

**Comment by sylvestre** [comment]
Impressive work.
Some issues however:
```
error[E0599]: no method named `strip_suffix` found for type `&str` in the current scope

   --> src/uu/basename/src/basename.rs:118:14

    |

118 |         name.strip_suffix(suffix).unwrap_or(name).to_owned()

    |              ^^^^^^^^^^^^ method not found in `&str`

error: aborting due to previous error
```

**Comment by rethab** [comment]
Should be fixed.

When is the minimum rust version updated? How does that work?

**Comment by sylvestre** [comment]
It is 1.40 currently.
We don't have a real policy. it is more when we have a bunch of new things interesting in a version (or an important crate)

See https://github.com/uutils/coreutils/pull/1909 for example

**Pull request merged**